### PR TITLE
Make API Key and Dataset optional

### LINKED ISSUE #83: APIKey and Dataset are Required but should be Optional
Right now we throw an exception if apikey or dataset are not set. There are situations when you may not want to set them such as when sending to an opentelemetry collectors. It would be better to throw a warning message instead.

### PR DESCRIPTION
**Pull request by vreynolds**
* downgrade missing apiKey/dataset to a warning
* closes #83

### CODE REVIEW

**Review by JamieDanielson — COMMENTED** [review]
System.out is used in Agent but not in SDK - is this intentional? Do we still want SDK to throw an exception (vs warn) for missing api key and dataset?

**Comment by vreynolds** [comment]
Yep, I left the logger in the SDK, since that wasn't causing issues, and it's a nicer way to signal than stdout. I think both SDK and agent should be consistent in their requirements.

File: sdk/src/main/java/io/honeycomb/opentelemetry/HoneycombSdk.java [hunk 1]
```diff
@@ -246,10 +247,19 @@ public HoneycombSdk buildAndRegisterGlobal() {
          * @see GlobalOpenTelemetry
          */
         public HoneycombSdk build() {
-            Preconditions.checkNotNull(apiKey, "apiKey must be non-null");
-            Preconditions.checkNotNull(dataset, "dataset must be non-null");
             Preconditions.checkNotNull(sampler, "sampler must be non-null");
 
+            Logger logger = Logger.getLogger(HoneycombSdk.class.getName());
+
+            if (apiKey == null) {
```
Comment (MikeGoldsmith) [inline sdk/src/main/java/io/honeycomb/opentelemetry/HoneycombSdk.java:254]: Same as above, I think we should check for null and empty strings here too.

File: agent/src/main/java/io/honeycomb/opentelemetry/HoneycombAgent.java [hunk 2]
```diff
@@ -12,45 +12,33 @@
...
     }
 
-    private static void configureEnvironment() throws Exception {
+    private static void configureEnvironment() {
+
         final String apiKey = EnvironmentConfiguration.getHoneycombApiKey();
         final String apiEndpoint = EnvironmentConfiguration.getHoneycombApiEndpoint();
         final String dataset = EnvironmentConfiguration.getHoneycombDataset();
 
         if (apiKey == null) {
```
Comment (MikeGoldsmith) [inline agent/src/main/java/io/honeycomb/opentelemetry/HoneycombAgent.java:30]: I think we should validate for null and empty strings.

**Comment by JamieDanielson** [comment]
> Yep, I left the logger in the SDK, since that wasn't causing issues, and it's a nicer way to signal than stdout. I think both SDK and agent should be consistent in their requirements.

You're right! I had something off on my side, this works out great now.

**Review by JamieDanielson — APPROVED** [review]
Using System.out on Agent and Logger on SDK for missing API key and Dataset now allows app to run without throwing exception. Also confirmed using a collector with those variables in place properly sends data to Honeycomb (WARNs seen in log as expected).

File: sdk/src/main/java/io/honeycomb/opentelemetry/HoneycombSdk.java (hunk 1 above)
Comment (vreynolds) [reply sdk/src/main/java/io/honeycomb/opentelemetry/HoneycombSdk.java]: There's definitely a lot we still _can_ do to validate things up front (we don't check that the URL is well-formed, for example). Added the empty string check, that was easy enough to add onto this.

**Review by MikeGoldsmith — APPROVED** [review]
Looks good. I agree there is more validation we could do, eg urls.

**Pull request merged**